test/memspaces/memspace_host_all.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "memory_target_numa.h"
+#include "memspace_helpers.hpp"
+#include "memspace_internal.h"
+#include "test_helpers.h"
+
+#include <numa.h>
+#include <numaif.h>
+#include <umf/memspace.h>
+
+using umf_test::test;
+
+TEST_F(numaNodesTest, memspaceGet) {
+    umf_memspace_handle_t hMemspace = umfMemspaceHostAllGet();
+    UT_ASSERTne(hMemspace, nullptr);
+
+    // Confirm that the HOST ALL memspace is composed of all available NUMA nodes.
+    UT_ASSERTeq(hMemspace->size, nodeIds.size());
+    for (size_t i = 0; i < hMemspace->size; i++) {
+        // NUMA memory target internally casts the config directly into priv.
+        struct umf_numa_memory_target_config_t *numaTargetCfg =
+            (struct umf_numa_memory_target_config_t *)hMemspace->nodes[i]->priv;
+        UT_ASSERT(std::find(nodeIds.begin(), nodeIds.end(),
+                            numaTargetCfg->id) != nodeIds.end());
+    }
+}
+
+TEST_F(memspaceHostAllTest, providerFromHostAllMemspace) {
+    umf_memory_provider_handle_t hProvider = nullptr;
+    enum umf_result_t ret =
+        umfMemoryProviderCreateFromMemspace(hMemspace, nullptr, &hProvider);
+    UT_ASSERTeq(ret, UMF_RESULT_SUCCESS);
+    UT_ASSERTne(hProvider, nullptr);
+
+    umfMemoryProviderDestroy(hProvider);
+}
+
+TEST_F(memspaceHostAllProviderTest, allocFree) {
+    void *ptr = nullptr;
+    size_t size = SIZE_4K;
+    size_t alignment = 0;
+
+    enum umf_result_t ret =
+        umfMemoryProviderAlloc(hProvider, size, alignment, &ptr);
+    UT_ASSERTeq(ret, UMF_RESULT_SUCCESS);
+    UT_ASSERTne(ptr, nullptr);
+
+    memset(ptr, 0xFF, size);
+
+    ret = umfMemoryProviderFree(hProvider, ptr, size);
+    UT_ASSERTeq(ret, UMF_RESULT_SUCCESS);
+}
+
+TEST_F(memspaceHostAllProviderTest, memoryPolicy) {
+    void *ptr = nullptr;
+    size_t size = SIZE_4K;
+    size_t alignment = 0;
+
+    enum umf_result_t umf_ret =
+        umfMemoryProviderAlloc(hProvider, size, alignment, &ptr);
+    UT_ASSERTeq(umf_ret, UMF_RESULT_SUCCESS);
+    UT_ASSERTne(ptr, nullptr);
+
+    auto it = std::max_element(nodeIds.begin(), nodeIds.end());
+    UT_ASSERT(it != nodeIds.end());
+
+    unsigned long maxNodeId = *it;
+
+    unsigned bitsPerUlong = sizeof(unsigned long) * 8;
```

Comment:
please add some comments here that would explain why we expect BIND mode etc

Reply:
Done.